Add missing <sys/sysmacros.h>

mmap_v1_engine.cpp is using major()/minor(), which are defined by
<sys/sysmacros.h>. This header file was implicitly included by
<sys/types.h> in glibc < 2.28. But this glibc 2.28, this is no longer
the case, so <sys/sysmacros.h> needs to be included explicitly.

Signed-off-by: Florian La Roche <F.LaRoche@pilz.de>

--- ./src/mongo/db/storage/mmap_v1/mmap_v1_engine.cpp
+++ ./src/mongo/db/storage/mmap_v1/mmap_v1_engine.cpp
@@ -32,6 +32,7 @@
 
 #include "mongo/db/storage/mmap_v1/mmap_v1_engine.h"
 
+#include <sys/sysmacros.h>
 #include <boost/filesystem/path.hpp>
 #include <boost/filesystem/operations.hpp>
 #include <fstream>
